### PR TITLE
AzureRM Debug

### PR DESCRIPTION
**Pull request by vincentdass**

### CODE REVIEW

File: Tasks/AzureRmWebAppDeployment/azurermutil.ts [hunk 1]
```diff
@@ -190,6 +197,5 @@ export async function getAzureRMWebAppConfigDetails(SPN, webAppName: string, res
             deferred.reject(tl.loc('ErrorOccurredStausCode0',response.statusCode));
```
Comment (Ajay-MS) [inline Tasks/AzureRmWebAppDeployment/azurermutil.ts]: add statusMessage also in same line

Comment (vincentdass) [inline Tasks/AzureRmWebAppDeployment/azurermutil.ts]: done

**Comment by Ajay-MS** [comment]
Can you remove legacy folder as part of this PR.

File: Tasks/AzureRmWebAppDeployment/kuduutility.ts [hunk 2]
```diff
@@ -8,6 +8,45 @@ var gulp = require('gulp');
 var zip = require('gulp-zip');
 var AdmZip = require('adm-zip');
 
+export async function appOffineKuduService(method: string, publishUrl: string, physicalPath: string, headers) {
```
Comment (Ajay-MS) [inline Tasks/AzureRmWebAppDeployment/kuduutility.ts]: argument should be a state not method type

File: Tasks/AzureRmWebAppDeployment/kuduutility.ts [hunk 3]
```diff
@@ -8,6 +8,45 @@ var gulp = require('gulp');
...
+    var kuduDeploymentURL = "https://" + publishUrl + "/api/vfs/" + physicalPath + '/app_offline.htm';
+    if(method === 'PUT') {
+        var offlineFilePath = path.join(tl.getVariable('System.DefaultWorkingDirectory'), 'app_offline.htm');
+        fs.writeFileSync(offlineFilePath, '<h1>The Web Page is temporarily unavailable !</h1>');
+        var webAppReadStream = fs.createReadStream(offlineFilePath);
+        httpObj.sendFile('PUT', kuduDeploymentURL, webAppReadStream, headers, (error, response, body) => {
+            if (error) {
+                defer.reject(error);
+            }
+            else if (response.statusCode === 201 || response.statusCode === 204) {
```
Comment (Ajay-MS) [inline Tasks/AzureRmWebAppDeployment/kuduutility.ts]: Check if there could be 200 status code also and mentioned it too.

Comment (vincentdass) [inline Tasks/AzureRmWebAppDeployment/kuduutility.ts]: done

File: Tasks/AzureRmWebAppDeployment/Strings/resources.resjson/en-US/resources.resjson [hunk 4]
```diff
@@ -68,5 +67,9 @@
   "loc.messages.UnabletofindthelocationofMSDeployfromregistryonmachineError": "Unable to find the location of MS Deploy from registry on machine (Error : %s)",
   "loc.messages.Nopackagefoundwithspecifiedpattern": "No package found with specified pattern",
   "loc.messages.MorethanonepackagematchedwithspecifiedpatternPleaserestrainthesearchpatern": "More than one package matched with specified pattern. Please restrain the search patern.",
-  "loc.messages.Trytodeploywebappagainwithappofflineoptionselected": "Try to deploy web app again with app_offline option selected."
+  "loc.messages.Trytodeploywebappagainwithappofflineoptionselected": "Try to deploy web app again with app_offline option selected.",
+  "loc.messages.AppOfflineModeenabled": "App Offline Mode enabled.",
+  "loc.messages.Failedtoenableappofflinemode": "Failed to enable app offline mode. Status Code: %s (%s)",
+  "loc.messages.AppOflineModedisabled": "App Ofline Mode disabled.",
```
Comment (kmkumaran) [inline Tasks/AzureRmWebAppDeployment/Strings/resources.resjson/en-US/resources.resjson]: Typo - offline.

**Pull request merged**